pypeit/scienceimage.py

```diff
@@ -702,14 +702,14 @@ def build_crmask(self, stack, ivar=None):
         varframe = utils.calc_ivar(ivar)
         saturation = self.spectrograph.detector[self.det-1]['saturation']
         nonlinear = self.spectrograph.detector[self.det-1]['nonlinear']
-        sigclip, objlim = self.spectrograph.get_lacosmics_par(proc_par,binning=self.binning)
+        #sigclip, objlim = self.spectrograph.get_lacosmics_par(proc_par,binning=self.binning)
```

Comment:
Hi @profxj can you explain how you now deal with this functionality using your new meta data construct? It seems like you need to dynamically update the parset depending on the file in question that you are dealing with in order to make these binning dependent adjustments to the CR finding parameters. But perhaps I'm being ignorant here, and missed something simple. In my opinion parsets should be immutable constructs and not change depending on which file is read in.

Reply:
Yes, there are now two calls using the `Spectrograph` to set PypeItPar:

- The call using the output from `default_pypeit_par()`
- A call using `config_specific_par` which takes in a science frame filename

Both are executed at the start of `PypeIt`.  Perhaps ideally they'd be done
in the same call.  Looking at it again, I think I could do that.  Doing so now..

But to finish the response to your question, `PypeItPar` remains immutable after
the initial call(s) to `Spectrograph`.  The advance here is that parameters can be
changed at the onset depending on the instrument configuration.

Reply:
Hi @profxj, 

Okay but as far as I can see binning is still its separate category not included in this parset. For X-shooter we have 1x1 binned arcs applied to data of arbitrary binning which is why I ask.